pkg/agent/agent.go

```diff
@@ -1171,17 +1181,26 @@ func (i *Initializer) initNodeLocalConfig() error {
 func (i *Initializer) initVMLocalConfig(nodeName string) error {
 	klog.InfoS("Initialize VM config", "ExternalNode", nodeName)
 	wait.PollImmediateUntil(time.Minute, func() (done bool, err error) {
-		_, err = i.crdClient.CrdV1alpha1().ExternalNodes(i.namespace).Get(context.TODO(), nodeName, metav1.GetOptions{})
+		en, err := i.crdClient.CrdV1alpha1().ExternalNodes(i.namespace).Get(context.TODO(), nodeName, metav1.GetOptions{})
 		if err != nil {
+			klog.InfoS("Failed! Initialize VM config, retry for", "ExternalNode", nodeName)
+			return false, nil
+		}
+		if len(en.Spec.Interfaces) == 0 {
+			klog.InfoS("No interfaces found for", "ExternalNode", nodeName)
 			return false, nil
 		}
+		// TODO: Handle multiple interfaces in ExternalNode
+		i.vmInterface = en.Spec.Interfaces[0].Name
```

Comment:
It is possible that Interface.Name is not set, then this might hit issues. @mengdie-song has made a change to add webhook ensure IP must be set in the ExternalNode.Interface. So I think a better solution is to get the network interface by IP. 

Besides, this is only for Windows, maybe move the function logic to agent_windows.go?

Reply:
Sure, will move this snippet to agent_windows.go

Reply:
Done